chips/rp2040/src/pwm.rs

```diff
@@ -0,0 +1,1254 @@
+    // Enable register
+    // This register aliases the CSR_EN bits for all channels.
+    // Writing to this register allows multiple channels to be enabled or disabled
+    // or disables simultaneously, so they can run in perfect sync.
+    en: ReadWrite<u32, CH::Register>,
+    // Raw interrupts register
+    intr: WriteOnly<u32, CH::Register>,
+    // Interrupt enable register
+    inte: ReadWrite<u32, CH::Register>,
+    // Interrupt force register
+    intf: ReadWrite<u32, CH::Register>,
+    // Interrupt status after masking & forcing
+    ints: ReadOnly<u32, CH::Register>,
+}
+
+#[derive(Clone, Copy)]
+/// Fractional clock divider running mode
+///
+/// Each channel can be configured to run in four different ways:
+///
+/// + Free running: The fractional clock divider is always enabled. In this mode,
+/// pins A and B are configured as output pins. In other modes, pin B becomes
+/// an input pin.
+/// + High: The fractional clock divider is enabled when pin B is high.
+/// + Rising: The fractional clock divider is enabled when a rising-edge is
+/// detected on pin B.
+/// + Falling: The fractional clock divider is enabled when a falling-edge
+/// is detected on pin B.
+pub enum DivMode {
+    FreeRunning,
+    High,
+    Rising,
+    Falling,
+}
+
+/// Channel identifier
+///
+/// There are a total of 8 eight PWM channels.
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum ChannelNumber {
+    Ch0,
+    Ch1,
+    Ch2,
+    Ch3,
+    Ch4,
+    Ch5,
+    Ch6,
+    Ch7,
+}
+
+const CHANNEL_NUMBERS: [ChannelNumber; NUMBER_CHANNELS] = [
+    ChannelNumber::Ch0,
+    ChannelNumber::Ch1,
+    ChannelNumber::Ch2,
+    ChannelNumber::Ch3,
+    ChannelNumber::Ch4,
+    ChannelNumber::Ch5,
+    ChannelNumber::Ch6,
+    ChannelNumber::Ch7,
+];
+
+/// Each GPIO pin can be configured as a PWM pin.
+/// The following table shows the mapping between GPIO pins and PWM pins:
+///
+/// | GPIO  | PWM |
+/// | ----- | --- |
+/// | 0     | 0A  |
+/// | 1     | 0B  |
+/// | 2     | 1A  |
+/// | 3     | 1B  |
+/// | 4     | 2A  |
+/// | 5     | 2B  |
+/// | 6     | 3A  |
+/// | 7     | 3B  |
+/// | 8     | 4A  |
+/// | 9     | 4B  |
+/// | 10    | 5A  |
+/// | 11    | 5B  |
+/// | 12    | 6A  |
+/// | 13    | 6B  |
+/// | 14    | 7A  |
+/// | 15    | 7B  |
+/// | 16    | 0A  |
+/// | 17    | 0B  |
+/// | 18    | 1A  |
+/// | 19    | 1B  |
+/// | 20    | 2A  |
+/// | 21    | 2B  |
+/// | 22    | 3A  |
+/// | 23    | 3B  |
+/// | 24    | 4A  |
+/// | 25    | 4B  |
+/// | 26    | 5A  |
+/// | 27    | 5B  |
+/// | 28    | 6A  |
+/// | 29    | 6B  |
+///
+/// **Note**:
+///
+/// + The same PWM output can be selected on two GPIO pins. The same signal will appear on each
+/// GPIO.
+/// + If a PWM B pin is used as an input, and is selected on multiple GPIO pins, then the PWM
+/// channel will see the logical OR of those two GPIO inputs
+impl From<RPGpio> for ChannelNumber {
+    fn from(gpio: RPGpio) -> Self {
+        match gpio as u8 >> 1 & 7 {
```

Comment:
```suggestion
        match gpio as u8 >> 1 & 0b111 {
```

nit, I just think this is a bit clearer

Reply:
Done.